Add namespace and guards to mathmlCoreExtras.css

eerii opened the pull request:
#### 5d81af3f80a0583e2107f811905c7fc7c6392292
<pre>
Add namespace and guards to mathmlCoreExtras.css
<a href="https://bugs.webkit.org/show_bug.cgi?id=302771">https://bugs.webkit.org/show_bug.cgi?id=302771</a>

Reviewed by Frédéric Wang.

Add missing parts to mathmlCoreExtras.css.

* Source/WebCore/css/mathmlCoreExtras.css:
(#if defined(ENABLE_MATHML) &amp;&amp; ENABLE_MATHML):
(semantics &gt; :not(:first-child)): Deleted.

Canonical link: <a href="https://commits.webkit.org/304207@main">https://commits.webkit.org/304207@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8800fb5da8e970684e22d4e90633be23ee1cab72

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134558 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7016 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45792 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142084 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86517 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/136428 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7619 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6876 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102844 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/70121 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137505 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5329 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120600 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83642 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5187 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2806 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/2688 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114418 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38741 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144778 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6692 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39323 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111246 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6766 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5615 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111525 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28365 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5021 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116875 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/60554 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6743 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35068 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6556 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70325 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6791 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6667 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fred-wang reviewed (approved):
Can you please add a WTP test for this regression?

I guess you can do a reftest checking that "Hello World!" is indeed displayed twice:

```html
<!DOCTYPE html>
<div id="container"></div>
<script>
  ["semantics", "maction"].forEach(tagName => {
    const nonMathMLNS = "http://www.w3.org/1999/xhtml";
    let el = document.createElement(tagName, nonMathMLNS);
    el.innerHTML = "<span>Hello</span> <span>World !</span>;
    container.appendChild(el);
  });
</script>
```

or it could be a testharness checking the display value of the "World !" spans.

fred-wang commented on LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/non-mathml-namespace-expected.txt:
```diff
@@ -0,0 +1,24 @@
+layer at (0,0) size 800x600
```
please remove this file

fred-wang replied:
(WPT refests due visual comparison by taking screenshots, the dumped render tree is something used by webkit internal tests)

eerii replied:
Huh, it seems that the tests [are failing](https://ews-build.s3-us-west-2.amazonaws.com/macOS-Sonoma-Release-WK2-Tests-EWS/5716fb03-36781-stress-mode/results.html) because there isn't a result. Should I add anything?

fred-wang replied:
I think WebKit uses -expected.html rather than -ref.html

Be sure to rename to -ref.html when exporting.

fred-wang commented on LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/non-mathml-namespace-ref.html:
```diff
@@ -0,0 +1,13 @@
+  <body>
+    <div>
+      <semantics><span>Hello</span> <span>World</span></semantics>
+      <maction><span>Hello</span> <span>World</span></maction>
```
mmh, the ref file should always render "World!" (even before your patch) otherwise the test would not allow to check the regression...

So I assume you would need to use span elements instead of semantics/maction elements here ?

eerii replied:
Of course, sorry about that.

eerii committed:
Add namespace and guards to mathmlCoreExtras.css
https://bugs.webkit.org/show_bug.cgi?id=302771

Reviewed by Frédéric Wang.

Add missing parts to mathmlCoreExtras.css.

* Source/WebCore/css/mathmlCoreExtras.css:
(#if defined(ENABLE_MATHML) && ENABLE_MATHML):
(semantics > :not(:first-child)): Deleted.

Canonical link: https://commits.webkit.org/304207@main
